### PR TITLE
Refactor: Improve code stability and fix bugs

### PR DESCRIPTION
**Pull request by morgoth95**

### CODE REVIEW

File: nebullvm/inference_learners/tensorflow.py [hunk 1]
```diff
@@ -25,14 +26,15 @@ def run(self, *input_tensors: tf.Tensor) -> Tuple[tf.Tensor, ...]:
         return res
 
     def save(self, path: Union[str, Path], **kwargs):
-        path = Path(path)
+        path = Path(path) / SAVE_DIR_NAME
+        os.makedirs(path, exist_ok=True)
```
Comment (valeriosofi) [inline nebullvm/inference_learners/tensorflow.py:30]: change os with pathlib

File: nebullvm/inference_learners/tvm.py [hunk 2]
```diff
@@ -18,6 +22,7 @@
 )
 from nebullvm.base import ModelParams, DeepLearningFramework
```
Comment (valeriosofi) [inline nebullvm/inference_learners/tvm.py:23]: move this import up

File: nebullvm/utils/feedback_collector.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+            "hardware_setup": self._hw_info,
+            "model_metadata": self._model_info,
+            "nebullvm_version": VERSION,
+        }
+        headers = {
+            "accept": "application/json",
+            "Content-Type": "application/json",
+        }
+        url = "https://nebuly.cloud/v1/store_nebullvm_results"
+        response = requests.post(
```
Comment (valeriosofi) [inline nebullvm/utils/feedback_collector.py:156]: What happens if the url is down? We should set the timeout argument, otherwise it will wait indefinitely

**Review by valeriosofi — CHANGES_REQUESTED** [review]

File: nebullvm/inference_learners/tensorflow.py (hunk 1 above)
Comment (morgoth95) [reply nebullvm/inference_learners/tensorflow.py]: solved in https://github.com/nebuly-ai/nebullvm/pull/74/commits/892c91bb8001698f6eb385215f68c3e7f91fff86

File: nebullvm/inference_learners/tvm.py (hunk 2 above)
Comment (morgoth95) [reply nebullvm/inference_learners/tvm.py]: solved in https://github.com/nebuly-ai/nebullvm/pull/74/commits/892c91bb8001698f6eb385215f68c3e7f91fff86

File: nebullvm/utils/feedback_collector.py (hunk 3 above)
Comment (morgoth95) [reply nebullvm/utils/feedback_collector.py:156]: solved in https://github.com/nebuly-ai/nebullvm/pull/74/commits/892c91bb8001698f6eb385215f68c3e7f91fff86

**Pull request merged**